lib/index.d.ts

```diff
@@ -195,7 +203,7 @@ export class Component<
    * For use inside view templates, to create a child Panel component nested under this
    * component, which will share its state object and update cycle.
    */
-  child(tagName: string, config?: object): VNode;
+  child<T = object>(tagName: string, config?: T): VNode;
```

Comment:
Drive-by type update to allow people to opt-in to better typing when using `this.child`